Load config from ssm

alexduf opened the pull request:
## What does this change?
 - load the configuration from SSM, using the simple-configuration library
 - refactor the AWS clients such that they use the same credentials
 - slight change of behaviour in our SQS logic, such that we create one client for the whole lifetime of the lambda

## How to test
Will be testable on CODE

## How can we measure success?
N/A

## Have we considered potential risks?
N/A

## Todo
 - [x] Add the configuration in SSM CODE
 - [x] Test in CODE
 - [x] Add the configuration in SSM PROD

_Note: We're currently awaiting for the production credentials_

buck06191 commented on src/main/scala/com/gu/newsletterlistcleanse/Env.scala:
```diff
@@ -8,5 +8,5 @@ object Env {
-    Option(System.getenv("Stage")).getOrElse("DEV"))
-}
\ No newline at end of file
+    Option(System.getenv("Stage")).getOrElse("CODE"))
```
Note to future developers: This was changed as there is no `DEV` stage used in this project. Having a `DEV` fallback leads to errors if the environment variable isn't set.

buck06191 reviewed (approved):
LGTM

alexduf commented:
I've now done the following:
 - Created the config in SSM
 - Adjusted the SSM loading path to match what the identity team is doing in their account
 - Gave the right to the lambdas to load the config from SSM
 - Fixed the `GetCutOffDatesLambdaInput` case class to be instantiated like a [standard Java Bean](https://en.wikipedia.org/wiki/JavaBeans), particularly Java Beans require a constructor with no argument. This is useful so that Jackson (the Json library AWS uses) can create an instance of the object.

It now runs in CODE, ready to be re-reviewed and merged :)

alexduf commented:
I've added the config in prod, pending the working service account, and the braze api key

buck06191 reviewed (approved):
![Gif of dogs clapping](https://media.giphy.com/media/kBZBlLVlfECvOQAVno/giphy.gif)

Good to go

alexduf commented:
I always like a good review gif 👍
